test/CQRS/LeanCode.CQRS.MassTransitRelay.Tests/Middleware/EventsPublisherMiddlewareTests.cs

```diff
@@ -0,0 +1,194 @@
+#nullable enable
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using FluentAssertions;
+using LeanCode.Contracts;
+using LeanCode.CQRS.Execution;
+using LeanCode.CQRS.MassTransitRelay.Middleware;
+using LeanCode.DomainModels.Model;
+using LeanCode.IntegrationTestHelpers;
+using MassTransit;
+using MassTransit.Testing;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+
+namespace LeanCode.CQRS.MassTransitRelay.Tests.Middleware;
+
+public sealed class EventsPublisherMiddlewareTests : IDisposable, IAsyncLifetime
+{
+    private readonly IHost host;
+    private readonly TestServer server;
+    private readonly IServiceProvider serviceProvider;
+    private readonly ITestHarness harness;
+
+    public EventsPublisherMiddlewareTests()
+    {
+        host = new HostBuilder()
```

Comment:
Do we need full web host to test this?

Reply:
Maybe we don't, but I feel using the webhost here is more handy